charts/prometheus-snmp-exporter/templates/_helpers.tpl

```diff
@@ -40,3 +40,17 @@ Create the name of the service account to use
     {{ default "default" .Values.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
+
+{{/* Generate basic labels */}}
+{{- define "prometheus-snmp-exporter.labels" }}
+app.kubernetes.io/name: {{ template "prometheus-snmp-exporter.name" . }}
+app.kubernetes.io/instance: "{{ .Release.Name }}"
+app.kubernetes.io/component: metrics 
+app.kubernetes.io/managed-by: "{{ .Release.Service }}"
+app.kubernetes.io/version: "{{ .Chart.AppVersion }}"
+app.kubernetes.io/part-of: {{ template "prometheus-snmp-exporter.name" . }} 
+helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
```

Comment:
Whats the benefit of adding all available labels here?

Imho, this should more look like as i the helm default which would look like this, if created with helm 3.6.0:

```
{{/*
Common labels
*/}}
{{- define "prometheus-snmp-exporter.labels" -}}
helm.sh/chart: {{ include "prometheus-snmp-exporter.chart" . }}
{{ include "prometheus-snmp-exporter.selectorLabels" . }}
{{- if .Chart.AppVersion }}
app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
{{- end }}
app.kubernetes.io/managed-by: {{ .Release.Service }}
{{- end }}

{{/*
Selector labels
*/}}
{{- define "prometheus-snmp-exporter.selectorLabels" -}}
app.kubernetes.io/name: {{ include "prometheus-snmp-exporter.name" . }}
app.kubernetes.io/instance: {{ .Release.Name }}
{{- end }}
``` 

Using selectorlabels seperatly would also enable you to use the variable in the service instead of the now hardcoded labels: https://github.com/prometheus-community/helm-charts/blob/8edbbe9c6ad80a4bcd7ca7856e303e78806fc19e/charts/prometheus-snmp-exporter/templates/service.yaml#L25-L27

Reply:
@monotek i split the labels.